MergerCli/utils/BatchStatusManager.cs

```diff
@@ -74,13 +152,42 @@ public bool IsBaseLayerNew()
 
         public string? GetLayerBatchIdentifier(string layer)
         {
-            if (this.States.ContainsKey(layer))
+            lock (_locker)
+            {
+                if (this.States.ContainsKey(layer))
+                {
+                    return this.States[layer].BatchIdentifier;
+                }
+                return null;
+            }
+        }
+
+        public long? GetTotalCompletedTiles(string layer)
+        {
+            lock (_locker)
             {
-                return this.States[layer].BatchIdentifier;
+                if (this.States.ContainsKey(layer))
+                {
+                    return this.States[layer].TotalCompletedTiles;
+                }
+                return null;
             }
-            return null;
         }
 
+        public void ResetBatchStatus()
+        {
+            foreach(string layer in this.States.Keys)
+            {
+                foreach (string batch in this.States[layer].Batches.Keys)
+                {
+                    if (batch is not null)
+                    {
+                        this.States[layer].Batches.TryUpdate(batch, false, true);
```

Comment:
Why not update as such: `this.States[layer].Batches[batch] = false`?
There is no reason to check the value, you want to just override the value.